File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int maxW = 0;
        int maxC = 0;
        for (int[] size: sizes) {
            int w = Math.max(size[0], size[1]);
            int c = Math.min(size[0], size[1]);
            maxW = Math.max(maxW, w);
            maxC = Math.max(maxC, c);
        }
        return maxW * maxC;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # 가로는 두 변 중 긴 것, 세로는 두 변 중 작은 것 넣어서 그 중 max끼리 곱
    maxW , maxH = 0, 0
    for s in sizes:
        width = max(s[0], s[1])
        height = min(s[0], s[1])
        maxW = max(width, maxW)
        maxH = max(height, maxH)
    return maxW * maxH